app/assets/javascripts/calendar/calendar.js

```diff
@@ -0,0 +1,61 @@
+$(function () {
+    console.log(events);
+
+   events.forEach(function (block) {
+        block.forEach(function (event) {
+            var tableCell = getEventTableCell(event);
+            var eventContainer = tableCell.children('.event-container');
+            setEventSize(eventContainer, block, event);
+
+            var length = eventLengthInSeconds(event);
+
+            $.ajax({
+                type: 'GET',
+                dataType: 'html',
+                url: eventUrl + '/' + event.id
+            }).done(function (data) {
+                eventContainer.append(data);
+                var height = length * parseInt(tableCell.css('height'));
+                getEventElement(eventContainer, event).css({height: height + 'px'});
+            });
+        });
+    });
+
+    $(window).resize(function () {
+        events.forEach(function (block) {
+            block.forEach(function (event) {
+                var tableCell = getEventTableCell(event);
+                var eventContainer = tableCell.children('.event-container');
+                setEventSize(eventContainer, block, event);
```

Comment:
Iteration over blocks & events and then some basic setup is duplicated here. Do you have any idea how to make it more DRY?